src/hibernate-resume/hibernate-resume.c

```diff
@@ -1,58 +1,76 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <errno.h>
-#include <stdio.h>
 #include <sys/stat.h>
 
-#include "alloc-util.h"
 #include "devnum-util.h"
 #include "fileio.h"
 #include "initrd-util.h"
 #include "log.h"
+#include "main-func.h"
+#include "parse-util.h"
+#include "static-destruct.h"
+#include "string-util.h"
 
-int main(int argc, char *argv[]) {
+static char *arg_resume_device = NULL;
+static uint64_t arg_resume_offset = 0;
+
+STATIC_DESTRUCTOR_REGISTER(arg_resume_device, unsetp);
+
+static int run(int argc, char *argv[]) {
         struct stat st;
-        const char *device;
         int r;
 
-        if (argc != 2) {
-                log_error("This program expects one argument.");
-                return EXIT_FAILURE;
-        }
-
         log_setup();
 
+        if (argc != 3)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "This program expects two arguments.");
+
         umask(0022);
 
-        /* Refuse to run unless we are in an initrd() */
         if (!in_initrd())
-                return EXIT_SUCCESS;
+                return 0;
 
-        device = argv[1];
+        arg_resume_device = argv[1];
 
-        if (stat(device, &st) < 0) {
-                log_error_errno(errno, "Failed to stat '%s': %m", device);
-                return EXIT_FAILURE;
-        }
+        r = safe_atou64(argv[2], &arg_resume_offset);
```

Comment:
honestly, i'd make the second arg optional and just assume it#s zero.

This doesn't matter much, given that this is an internal command. but who knows, maybe someone glued together shell scripts that call the command directly. given it's easy we can just support the old command line, and just imply zero for the offset if only the device is specified.